novikov_dmitry/main.cpp

```diff
@@ -1,37 +1,147 @@
+    if (current_depth != graph.get_depth()) {
+      log_string << ", ";
+    }
+  }
+  log_string << "],\n";
+  log_string << "  edges: " << graph.get_edge_map().size() << ", {";
+  const auto& count_edges_of_color_map = set_count_edges_of_color(graph);
+  for (int color_index = 0; color_index < count_edges_of_color_map.size();
+       ++color_index) {
+    log_string << uni_cpp_practice::color_to_string(
+                      count_edges_of_color_map[color_index].first)
+               << ": " << count_edges_of_color_map[color_index].second;
+    if (color_index + 1 != count_edges_of_color_map.size()) {
+      log_string << ", ";
+    }
+  }
+  log_string << "}\n}\n";
+  return log_string.str();
+}
+}  // namespace logger
+
+const uni_cpp_practice::Depth handle_depth_input() {
+  uni_cpp_practice::Depth
+      depth;  //Глубина графов (int от 0 и до бесконечности).
+  std::cout << "Enter the Depth of the graphs" << std::endl;
   std::cin >> depth;
   while (depth < 0) {
     std::cout << "Try to enter a natural number or zero" << std::endl;
     std::cin >> depth;
   }
-  std::cout << "Enter the Number of new vertices from each vertex of the graph"
+  return depth;
+}
+int handle_new_vertices_num_input() {
+  int new_vertices_num;  //Количество вершин, генерируемых каждой родительской
+                         //вершиной.
+  std::cout << "Enter the Number of new vertices from each vertex of the graphs"
             << std::endl;
   std::cin >> new_vertices_num;
   while (new_vertices_num < 0) {
     std::cout << "Try to enter a natural number or zero" << std::endl;
     std::cin >> new_vertices_num;
   }
-  std::stringstream ss_filename;
-  ss_filename << dafault_filename << dafault_fileformat;
-  const auto graph = graph_generation::generate_graph(depth, new_vertices_num);
-  std::ofstream file_out;
-  file_out.open(ss_filename.str(), std::fstream::out | std::fstream::trunc);
-  file_out << graph.to_string();
-  file_out.close();
+  return new_vertices_num;
+}
+
+int handle_graphs_count_input() {
+  int graphs_count;  // Количество генерируемых графов.
+  std::cout << "Enter the Count of new graphs" << std::endl;
+  std::cin >> graphs_count;
+  while (graphs_count <= 0) {
+    std::cout << "Try to enter a natural number" << std::endl;
+    std::cin >> graphs_count;
+  }
+  return graphs_count;
+}
 
+int main() {
+  const std::string dafault_filename = "Graph";
+  const std::string dafault_fileformat = ".json";
+
+  const int depth = handle_depth_input();
+  const int new_vertices_num = handle_new_vertices_num_input();
+  const int graphs_count = handle_graphs_count_input();
+
+  try {
+    if (!std::filesystem::create_directory("./temp")) {
+      std::cout << "A directory wasn't created\n";
+    }
+  } catch (const std::exception& ex) {
+    std::cout << ex.what() << "\n";
+  }
+
+  auto& logger = uni_cpp_practice::Logger::get_instance();
+  logger.set_file("./temp/log.txt");

```

Comment:
Усложняет чтение `main()` и не добавляет полезной информации - это детали имплементации, лучше тоже вынести в отдельную функцию, например
```suggestion
  auto& logger = prepare_logger();
```